packages/py-ab-testing/ABTesting/controller.py

```diff
@@ -19,15 +25,23 @@ def match_user_cohort(
     # type: (Dict, Union[str, int], Dict[str, str]) -> str
     user_segment_num = get_modulo_value(experiment_config['name'], user_id)
     allocated_cohort = 'control'
+    within_allocation_range = False
+
     for cohort in experiment_config['cohorts']:
         for force_include_key, force_include_val in cohort.get('force_include', {}).items():
             if force_include_key in user_profile and user_profile[force_include_key] in force_include_val:
                 return cohort['name']
+
         if allocated_cohort == 'control':
             for allocation in cohort.get('allocation', []):
                 if allocation[0] <= user_segment_num < allocation[1]:
-                    allocated_cohort = cohort['name']
+                    within_allocation_range = True
                     break
+
+            if(within_allocation_range):
+                if(validate_criteria(cohort.get('allocation_criteria', []), user_profile)):
```

Comment:
isn't it supposed to be `cohort.get('allocation_criteria', {})`

Reply:
@ZigZagT you are right! Updated!